crates/iceberg/src/arrow/schema.rs

```diff
@@ -741,7 +766,7 @@ macro_rules! get_parquet_stat_as_datum {
                     };
                     Some(Datum::new(
                         primitive_type.clone(),
-                        PrimitiveLiteral::Int128(i128::from_be_bytes(bytes.try_into()?)),
+                        PrimitiveLiteral::Int128(i128::from_be_bytes(extend_to_i128_big_endian(bytes.into())?)),
```

Comment:
I revisited the code and found that the root cause might be that we didn't handle the decimal bytes data correctly. By design, Iceberg only keeps the smallest bits of the decimal. To decode it properly, we may need to follow the same logic outlined here:

https://github.com/apache/iceberg-rust/blob/09fa1fa19488ab63c042aa692ac730fe1a7dc987/crates/iceberg/src/spec/values.rs#L436-L445

Reply:
Thanks! Nice catch